finals-2024-01-16/code/5332-milana-sokolova/src/app/statistics/statistics.component.ts

```diff
@@ -0,0 +1,37 @@
+// statistics.component.ts
+
+import { Component, OnInit } from '@angular/core';
+import { BandsService } from '../bands.service';
+
+@Component({
+  selector: 'app-statistics',
+  templateUrl: './statistics.component.html',
+  styleUrls: ['./statistics.component.css']
+})
+export class StatisticsComponent implements OnInit {
+  totalBands: number = 0;
+  totalBandMembers: number = 0;
+  totalGenres: number = 0;
+  totalAlbums: number = 0;
+  bandsPerGenre: { genre: string, count: number }[] = [];
+  albumsPerGenre: { genre: string, count: number }[] = [];
+  albumsPerDecade: { decade: string, count: number }[] = [];
+  citiesWithoutBands: number = 0;
+  citiesWithoutBandsList: string[] = [];
+
+  constructor(private bandsService: BandsService) { }
+
+  ngOnInit(): void {
+    this.bandsService.getStatistics().subscribe(statistics => {
```

Comment:
**If** we had a functional service, and **if** that service had this method, it would be just what I had in mind. Like this, +2 for the component.